application/classes/Logger.php

```diff
@@ -9,41 +9,46 @@ public static function instance() {
 	}
 	
 	private static function loadLogger() {
-		$prov_config = Kohana::$config->load('logger');
+		if (is_null(Kohana::$config)) {
+			$prov_config = [ 'type' => 'syslog', 'config' => 'controll' ];
+		} else {
+			$prov_config = Kohana::$config->load('logger');
+		}
 		$fullclass = "Log_" . str_replace('/', '_', ucfirst($prov_config['type']));
-		$logger = new $fullclass($prov_config);
-		Kohana::$log->attach($logger);
+		$logger = new $fullclass(array_key_exists('config', $prov_config) ? $prov_config['config'] : $prov_config);
+		if (!is_null(Kohana::$log))
+			Kohana::$log->attach($logger);
 		return $logger;
 	}
 
 	public static function debug($message, array $context = NULL, array $additional = NULL) {
 		self::instance();
-		return Kohana::$log->add(Kohana_Log::DEBUG, $message, self::parseContext($context), $additional);
+		return (Kohana::$log ?? Log::instance())->log(Kohana_Log::DEBUG, $message, self::parseContext($context));
```

Comment:
The problem I've discussed above was a crash that needed to be reported but happened before Kohana::$log was inited. This was a really edge case that happened due to some bad code on my side - I maybe just need to remove this entire patch.